packages/ra-ui-materialui/src/list/datagrid/Datagrid.tsx

```diff
@@ -35,8 +35,14 @@ import { DatagridClasses, DatagridRoot } from './useDatagridStyles';
 import { BulkActionsToolbar } from '../BulkActionsToolbar';
 import { BulkDeleteButton } from '../../button';
 import { ListNoResults } from '../ListNoResults';
-
-const defaultBulkActionButtons = <BulkDeleteButton />;
+import { SelectAllButton } from '../../button/SelectAllButton';
+
+const defaultBulkActionButtons = (
+    <>
+        <SelectAllButton />
```

Comment:
No, you must override the BulkActions Toolbar commponent as Selectall isn't a bulk action button ber se, but a feature of the toolbar. 

You must also introduce a new prop to Datagrid to let use pass a custom bulk action toolbar component.